nucleus/base/models.py

```diff
@@ -0,0 +1,102 @@
+import json
+from datetime import timedelta
+from logging import getLogger
+
+from django.conf import settings
+from django.db import models
+from django.db.models.signals import post_save, m2m_changed
+from django.dispatch import receiver
+from django.utils.timezone import now
+
+from crum import get_current_user
+from django_extensions.db.fields import CreationDateTimeField
+from spinach.contrib.spinachd.tasks import tasks
+
+
+log = getLogger(__name__)
+M2M_ACTIONS = ['post_add', 'post_remove', 'post_clear']
+DEFAULT_BRANCH = settings.GITHUB_OUTPUT_BRANCH
+
+
+def send_instance_to_github(instance, branch=DEFAULT_BRANCH):
+    log.debug(f'send_instance_to_github, {instance._meta}, {instance.pk}')
+    author = get_current_user()
+    # give it a few seconds to be sure the db
+    # transaction is committed
+    tasks.schedule_at('save_to_github',
+                      now() + timedelta(seconds=5),
+                      str(instance._meta),
+                      instance.pk,
+                      author.pk,
+                      branch)
+
+
+@receiver(post_save)
```

Comment:
I think it's good practice to set `dispatch_id` to avoid duplicate signal registration
https://docs.djangoproject.com/en/3.0/topics/signals/#preventing-duplicate-signals